### ISSUE TITLE
There is no way to catch an error while in a transaction

### ISSUE THREAD

**Issue by alextekartik** [powersync-ja/sqlite_async.dart#21]
I tried the following unit test (Added in basic_test.dart for a quick setup):

```dart
test('Catch error in transaction', () async {
  final db = await setupDatabase(path: path);
  await createTables(db);
  // Ok can catch here
  try {
    await db.getOptional('SELECT description FROM dummy1');
  } catch (_) {}
  await db.readTransaction((tx) async {
    // Not ok, cannot catch the error here
    try {
      await tx.getOptional('SELECT description FROM dummy2');
    } catch (_) {}
    await tx.getOptional('SELECT description FROM test_data');
  });
}, solo: true);
```

Even though I do catch the exception, the test fails.
It seems that once an error occurred there is no way to continue the transaction if we catch the exception. I guess this is by design but I think it prevents developers to catch error dynamically (such as a constraint error) while in a transaction to perform a different action in the same transaction.

**Comment by rkistner** [comment]
With SQLite, some errors can cause a transaction to be rolled back automatically. The issue is the current implementation assumes _any_ error can cause the transaction to be rolled back, and don't allow you to continue the transaction.

I'll have to check if there's a way to determine whether or not the transaction has been rolled back. As far as I can see, the standard way to do this is using the [sqlite3_get_autocommit()](https://www.sqlite.org/c3ref/get_autocommit.html) function, which is not exposed in the [sqlite3 bindings](https://github.com/simolus3/sqlite3.dart) yet.

**Comment by rkistner** [comment]
Fixed in v0.6.0
